### PR TITLE
Update docs for Identity Preview 4

### LINKED ISSUE #6781: Release Azure.Identity Library Preview 4 for Python
**API Design**
- [x] **Preview 4** Review Exception Patterns across languages (@schaabs)
- [x] **Preview 4** Review Authority Host Configuration and ensure consistency across Languages (@schaabs)

**Feature Work**
- [x] **Preview 4**  Authorization Code Flow support with MSAL https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-auth-code-flow
- [x] **Preview 4** Create a combination of Automated and Manual testing (postman, curl, etc documented steps)
   - [x] Define Positive/Negative/Stress SSO test cases (cross applications, cross browser tabs)

**Documentation**
- [x] N/A

**CUT**
- [x] **Preview 4** Design solution for publishing credentials to shared token cache for dev-time experience (@schaabs)
- [x] **Preview 4** Implement solution for publishing credentials to shared token cache
   - [x] Cross-Language-SDK testing to ensure the Shared Token Cache functions as expected (correct synchronization)

**Related**
   - [ ] Preview 4 for Python https://github.com/Azure/azure-sdk-for-python/issues/6781
   - [x] Preview 3 for Python https://github.com/Azure/azure-sdk-for-python/issues/6591
   - [x] Preview 2 for Python https://github.com/Azure/azure-sdk-for-python/issues/6337
   - [x] Preview 1 for Python https://github.com/Azure/azure-sdk-for-python/issues/5048

### PR DESCRIPTION
**Pull request by chlowell**
Closes #6781

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: sdk/identity/azure-identity/azure/identity/_credentials/environment.py [hunk 1]
```diff
@@ -75,6 +78,7 @@ def get_token(self, *scopes, **kwargs):  # pylint:disable=unused-argument
         # type: (*str, **Any) -> AccessToken
         """
         Request an access token for `scopes`.
+
```
Comment (lmazuel) [inline sdk/identity/azure-identity/azure/identity/_credentials/environment.py]: Side note, in theory the summary should be right after the """, without a new line:
https://www.python.org/dev/peps/pep-0257/#multi-line-docstrings

```
Multi-line docstrings consist of a summary line just like a one-line docstring, followed by a blank line, followed by a more elaborate description. The summary line may be used by automatic indexing tools; it is important that it fits on one line and is separated from the rest of the docstring by a blank line. The summary line may be on the same line as the opening quotes or on the next line. 
```

Comment (chlowell) [reply sdk/identity/azure-identity/azure/identity/_credentials/environment.py]: Thanks for pointing this out, I'll reformat everything before the next release (#7663)

**Pull request merged**